stl/src/filesystem.cpp

```diff
@@ -657,11 +657,15 @@ __std_win_error __stdcall __std_fs_get_file_id(__std_fs_file_id* const _Id, cons
     }
 
     _Last_error = __std_win_error{GetLastError()};
-    if (_Last_error != __std_win_error::_Invalid_parameter && _Last_error != __std_win_error::_Invalid_function) {
+    switch (_Last_error) {
+    case __std_win_error::_Invalid_parameter: // Older Windows versions
+    case __std_win_error::_Invalid_function: // Windows 10 1607
+    case __std_win_error::_Not_supported: // POSIX delete not supported by the file system (e.g. DevCom-857535)

```

Comment:
I'm strongly opposed to citing bugs in permanent comments (i.e. non-TRANSITION) in product code, because they become incomprehensible as bug databases rot. (Consider how useful a comment citing VSWhidbey-12345 would be today.) I believe that this should say "e.g. ReFS" or something similar.

Reply:
I don't think adding a "this might be a helpful bug number" for a comment that otherwise stands on its own is harmful; bug number references are only bad when the comment makes no sense without them.

That said, I removed the comment.

Reply:
+1 to Billy's rationale. It's better for a comment to potentially become useless than for it to potentially become misleading and incorrect. Conversely, I wouldn't oppose putting "ReFS" in the comment if we had test coverage to ensure that the comment would be fixed if ReFS starts to support POSIX delete.